terraform/roles.tf

```diff
@@ -288,3 +288,46 @@ resource aws_iam_role_policy_attachment ec2_readonly {
   role       = aws_iam_role.caascad_operator.name
   policy_arn = "arn:aws:iam::aws:policy/AmazonEC2ReadOnlyAccess"
 }
+
+resource aws_iam_role lambda-promtail {
+  name = "caascad-lambda-promtail-role"
+
+  assume_role_policy = jsonencode({
+    Version = "2012-10-17",
+    Statement = [{
+      Action = "sts:AssumeRole"
+      Effect = "Allow"
+      Principal = {
+        Service = "lambda.amazonaws.com"
+      }
+    }]
+  })
+
+  tags = {
+    caascad = "true"
+  }
+}
+
+data aws_iam_policy_document lambda-promtail {
+  statement {
+    actions = [
+      "logs:CreateLogStream",
+      "logs:PutLogEvents",
+    ]
+    resources = [
+      "arn:aws:logs:*:*:*",
+    ]
```

Comment:
Shouldn't we limit theses actions with `caascad-managed` tag ?

Reply:
@eonpatapon log Streams and groups dot not support condition keys. See [this page](https://docs.aws.amazon.com/service-authorization/latest/reference/list_amazoncloudwatchlogs.html#amazoncloudwatchlogs-log-group)

The only way would be to indicate a prefix in the resource like ""arn:aws:logs:*:*:caascad-*".

@mvienne-orange : I think logs:CreateLogGroup is missing.
In addition, your lambda should get log event? So logs:GetLogEvents and logs:GetLogRecord seem necessary.

Reply:
@yesteph we don't need logs:CreateLogGroup, because our Terraform configuration takes care of creating the log group, so Lambda does not need to. We decided to do this to link the lifecycle of the log group generated by the lambda to the terraform configuration. Moreover, logs:GetLogEvents and logs:GetLogRecord are not necessary, because these are granted by the aws_lambda_permission resource.

Reply:
@yesteph the page you linked says that there is no service-specific context-keys, but global context keys such as `aws:ResourceTag` are supported.
@eonpatapon done